File: iOS/CA2D/Graphics/Vertex.swift
import GLKit

struct Vertex {
    static let size = sizeof(Float) * 7

    let position: GLKVector3
    let color: GLKVector4

    var v: [Float] {
        return [
            position.x,
            position.y,
            position.z,
            color.r,
            color.g,
            color.b,
            color.a,
        ]
    }
}

File: Android/app/src/main/kotlin/cc/clv/android/ca2d/graphics/Vertex.kt
package cc.clv.android.ca2d.graphics


data class Vertex(val position: Position, val color: Color) {
    companion object {
        val size = Position.size + Color.size
    }

    val v: FloatArray
        get() = floatArrayOf(
                position.x,
                position.y,
                color.r,
                color.g,
                color.b
        )
}
